src/main.ts

```diff
@@ -81,6 +81,19 @@ export function buildObject(adapterInstance: ioBroker.Adapter, options: BuildObj
 		definition.common.desc = options.description;
 	}
 
+	if (definition.type === "state"){
+		if (Array.isArray(definition.common.type)){
+			if (options.value && definition.common.type.includes(typeof(options.value)) === true){
+				definition.common.type = typeof(options.value) as ioBroker.CommonType;
```

Comment:
`typeof` and `ioBroker.CommonType` are not the same.
* `typeof [] === "object"`, but the common.type would be `"array"`
* `typeof {} === "object"`, but the common.type would be `"object"`
* `typeof null === "object"`, but the common.type could be any one of the allowed types